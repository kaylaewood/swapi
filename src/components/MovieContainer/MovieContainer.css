.movie-container {
  display: flex;
  margin-top: 3vw;
  overflow: auto;
}
