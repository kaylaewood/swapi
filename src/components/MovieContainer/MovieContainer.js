import React from 'react';
import './MovieContainer.css';
import MovieCard from '../MovieCard/MovieCard';

function MovieContainer({ movies }) {
  const renderCards = () => {
    return movies.sort((a, b) => a.episode_id - b.episode_id).map(movie => {
      return (
        <MovieCard movie={movie} />
      )
    })
  }

  return (
    <section className='movie-container'>
      {renderCards()}
    </section>
  );
}

export default MovieContainer;
