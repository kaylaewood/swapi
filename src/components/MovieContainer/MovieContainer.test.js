import React from 'react';
import ReactDOM from 'react-dom';
import MovieContainer from './MovieContainer';
import { shallow } from 'enzyme';

describe('MovieContainer', () => {
  let movies = [
    {name: 'Titanic', episode_id: 1, characters: ['url1', 'url2']},
    {name: 'Alien', episode_id: 2, characters: ['url1', 'url2']},
    {name: 'Robocop', episode_id: 3, characters: ['url1', 'url2']}
  ]
  it('should match the snapshot', () => {
    const wrapper = shallow(<MovieContainer
      movies={movies}
    />)
    expect(wrapper).toMatchSnapshot();
    })
})
