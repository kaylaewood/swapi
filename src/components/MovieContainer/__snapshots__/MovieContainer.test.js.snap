// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer should match the snapshot 1`] = `
<section
  className="movie-container"
>
  <MovieCard
    key="1"
    movie={
      Object {
        "characters": Array [
          "url1",
          "url2",
        ],
        "episode_id": 1,
        "name": "Titanic",
      }
    }
  />
  <MovieCard
    key="2"
    movie={
      Object {
        "characters": Array [
          "url1",
          "url2",
        ],
        "episode_id": 2,
        "name": "Alien",
      }
    }
  />
  <MovieCard
    key="3"
    movie={
      Object {
        "characters": Array [
          "url1",
          "url2",
        ],
        "episode_id": 3,
        "name": "Robocop",
      }
    }
  />
</section>
`;
